apps/system/utils/utils_pminfo.c

```diff
@@ -0,0 +1,222 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <stdlib.h>
+#include <ctype.h>
+#if !defined(CONFIG_FS_AUTOMOUNT_PROCFS)
+#include <errno.h>
+#include <stdbool.h>
+#include <sys/mount.h>
+#endif
+#include <tinyara/sched.h>
+#include <tinyara/fs/fs.h>
+
+#include "utils_proc.h"
+
+#define PM_BUFLEN 128
+#define PM_BUFSIZE 30
+
+static bool is_numeric_name(const char *name)
+{
+	int name_index;
+
+	if (name == NULL) {
+		return false;
+	}
+
+	for (name_index = 0; name_index < CONFIG_NAME_MAX && name[name_index] != '\0'; name_index++) {
+		if (!isdigit(name[name_index])) {
+			/* Name contains something other than a numeric character */
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static int is_include_space(const char* buf)
+{
+	const char* local = buf;
+	int ret = ERROR;
+	while (*local) {
+		if (*local == ' ') {
+			ret = OK;
+			break;
+		}
+		local++;
+	}
+	return ret;
+}
+
+static void print_line(int loop)
+{
+	int i;
+
+	printf("\n---------------------");
+	for (i = 0; i < loop; i++) {
+		printf("----------------------");
+	}
+}
+
+static void pminfo_print_values(char *buf)
+{
+	int i;
+	int j;
+	char* pm_state[5] = {
+		"NORMAL",
+		"IDLE",
+		"STANDBY",
+		"SLEEP",
+		"COUNT"
+	};
+	pm_data pm_info[PM_BUFSIZE] = { NULL, };
+	pm_info[0] = buf;
+	int driver_cnt = 0;
+	bool flag = false;
+
+	for (i = 0; is_include_space(pm_info[i]) == OK; i++) {
+		pm_info[i] = strtok_r(pm_info[i], " ", &pm_info[i + 1]);
+	}
+
+	for (i = 0; pm_info[i]; i = i + 3 + driver_cnt) {
+
+		if (!strncmp(pm_info[i], "END", sizeof("END"))) {
+			break;
+		}
+		driver_cnt = atoi(pm_info[i]);
+
+		if (!flag) {
+			printf(" STATE  | STAY(SEC) ");
+			for (j = 0; j < (driver_cnt / 2); j++) {
+				printf("| drv_name , activity ");
+			}
+			printf("|");
+			flag = true;
+		}
+
+		print_line(driver_cnt / 2);
+
+		printf("\n%7s | %9s |", pm_state[atoi(pm_info[i + 1])], pm_info[i + 2]);
+		for (j = i + 3; j < i + 3 + driver_cnt; j += 2) {
+			printf(" %8s , %8s |", pm_info[j], pm_info[j + 1]);
+		}
+	}
+
+	if (flag) {
+		print_line(driver_cnt / 2);
+		printf("\n");
+	}
+}
+
+static int pminfo_read_proc(FAR struct dirent *entryp, FAR void *arg)
+{
+	int ret;
+	char *filepath;
+	char buf[PM_BUFLEN];
+
+	asprintf(&filepath, "%s/%s/%s", PROCFS_MOUNT_POINT, entryp->d_name, "pminfo");
+	ret = utils_readfile(filepath, buf, PM_BUFLEN, pminfo_print_values);
+	if (ret < 0) {
```

Comment:
If you assign the initial `ret` to `OK` and change the code like below, you can reduce the code.